### PR TITLE
[JENKINS-XXXXX] Added a dependency with blueocean-common

### PR DESCRIPTION
**Pull request by recena**
# Description

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

if `jenkins-design-language` does not have any Maven dependency with core plugins of Blue Ocean, this plugin could be installed independently of the rest of Blue Ocean plugins.

This would be an not desired behavior.

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: jenkins-design-language/pom.xml [hunk 1]
```diff
@@ -42,17 +50,4 @@
         </resources>
     </build>
 
-    <repositories>
```
Comment (recena) [inline jenkins-design-language/pom.xml]: It is not needed since it is defined in the parent.

File: pom.xml [hunk 2]
```diff
@@ -143,13 +143,13 @@
   <repositories>
     <repository>
       <id>repo.jenkins-ci.org</id>
-      <url>http://repo.jenkins-ci.org/public/</url>
+      <url>https://repo.jenkins-ci.org/public/</url>
```
Comment (recena) [inline pom.xml]: It is a good practice.

File: jenkins-design-language/pom.xml [hunk 3]
```diff
@@ -15,6 +15,14 @@
 
     <url>https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+Design+Language</url>
 
+    <dependencies>
+        <dependency>
+           <groupId>io.jenkins.blueocean</groupId>
+           <artifactId>blueocean-commons</artifactId>
```
Comment (kzantow) [inline jenkins-design-language/pom.xml]: Why is this added? JDL should not depend on this.

Comment (recena) [reply jenkins-design-language/pom.xml]: @kzantow 

> if jenkins-design-language does not have any Maven dependency with core plugins of Blue Ocean, this plugin could be installed independently of the rest of Blue Ocean plugins. This would be an not desired behavior.

If this plugin is completely independent and has its own lifecycle, please move it out side of this multi module project and use a version naming not synced with Blue Ocean.

Comment (vivek) [reply jenkins-design-language/pom.xml]: @recena Do not get this change. Yes this plugin has no dependency on other modules but its tied to blueocean release lifecycle and its intentional. 

>please move it out side of this multi module project and use a version naming not synced with Blue Ocean.

What motivates this change request?

Comment (michaelneale) [reply jenkins-design-language/pom.xml]: @recena not sure if depending on commons will solve anything - it will just get people to install more unrelated plugins than just jdl

**Comment by michaelneale** [comment]
the other changes other than the commons dep look great. 

@recena there are many other examples of "leaf" node plugins that may not necessarily make sense on their own (adding some behavior that doesn't make sense on its own - eg some workflow/pipeline ones, library plugins, and more)

**Comment by vivek** [comment]
Closing this, not sure if its relevant.
